libraries/Microsoft.Bot.Builder.Azure/AzureBlobStorage.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+using Microsoft.Bot.Builder.Core.Extensions;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Azure
+{
+    /// <summary>
+    /// Models IStorage around a dictionary 
+    /// </summary>
+    public class AzureBlobStorage : IStorage

```

Comment:
A BIG and detailed comment here, around the strategy of how this deals with blobs. Is all state for a bot in a single blob that's read/written on every call? Is this multiple blobs? One per key? 

Details here are critical, as they'll show up in our documentation.